### DISCUSSION TITLE
Next SSG does't provide sharing or storing data possibilities inside getStaticProps

### DISCUSSION THREAD

**Discussion by robertovg** [vercel/next.js#23937] [Ideas]
# Bug report

## Describe the bug

I have a some projects with next.js `9.4.4` and `9.4.5-canary` with SSG using `getStaticProps` fetching data from external Headless CMS. 
My question is how to share footer and header content on those pages. I know `_app` doesn't have `getStaticProps` and I don't see any good way to do that. Right now I'm just calling the same `getFooter()`, `getHeader()` in every page. But it means I'm querying the same data over and over again per page as I can see in the logs.
Additionally, if want to secure my data with authentication with a JWT I need to authorize once per page as I don't see any way to share the token between the `getStaticProps` calls, or a place where prepare this token as the first step of the build process. So 

- **I would like to get some "official" or good practices on how to resolve this problem.**
- If I need to authenticate before getting the content, I don't see any place where to store common data between the different calls. **I would appreciate a way to share information between different getStaticProps calls**. They seem to run in parallel with no way to share information like JWT between then, so it means I need to call to get a new JWT token per page with `getStaticProps` and this is not convenient.

## To Reproduce

Create a new next project with more than two pages with `getStaticProps` and try to share data (Header and Footer) between them.

## Expected behavior

Official guidelines or good practices on how to do it in Next.js

**Comment by balazsorban44** [comment]
This might be interesting to you: https://github.com/vercel/next.js/discussions/10949#discussioncomment-1203

If you scroll down, there might be some feasible options until `getStaticProps` in `_app` is added.
